data_steward/resources.py

```diff
@@ -96,10 +99,37 @@ def achilles_index_files():
     return achilles_index_files
 
 
-def fields_for(table):
-    json_path = os.path.join(fields_path, table + '.json')
+def fields_for(table, sub_path=None):
+    """
+    Return the json schema for any table identified in the fields directory.
+
+    Uses os.walk to traverse subdirectories
+
+    :param table: The table to get a schema for
+    """
+    path = os.path.join(fields_path, sub_path if sub_path else '')
+
+    # default setting
+    json_path = os.path.join(path, table + '.json')
+
+    unique_count = 0
+    for dirpath, _, files in os.walk(path):
+        if sub_path and os.path.basename(sub_path) != os.path.basename(dirpath):
```

Comment:
I'm not sure what the intent is here- is this to ensure that `sub_path` leads directly to the desired file (by ceasing traversal)? It seems that this would continue walking, but only when directories with the same name are nested e.g. `fields_for('person', 'foo/bar')` could be used to retrieve `foo/bar/bar/bar/person.json`.

I'm slightly conflicted as to whether this should `os.walk` even when `sub_path` is provided or if it should just assume `sub_path` points directly to the file; both approaches might potentially be useful. The docstring could indicate more explicitly the behavior when `sub_path` is provided. Theoretically both strategies could be supported [in the future].

Reply:
@mark-velez Yes.  This is exactly why I wanted to talk to you.  I wasn't sure if it made sense to let the function act one way with one set of parameters and another way with a different set of parameters.  

I don't like the idea of the nested directories, `foo/bar/bar/bar/person.json`.  My intention was to avoid finding `fields/person.json` if `table='person' and sub_path='foo'`.  In this case, more than one `person.json` file exists but we are specifically interested in finding the person schema at `fields/foo/person.json`.

With that thought, I suppose the `sub_path` parameter makes it explicit where we want to find the schema.  I'll move in the direction of:  if a `sub_path` is specified, that is the only directory that is traversed.

Reply:
_It could_ make sense to choose a search strategy based on params, but my preference would be to add that level of complexity only after a clear need is demonstrated. I think your choice should work well with existing usages. 👍